### PR TITLE
Make player walkthrough optional

### PR DESCRIPTION
**Pull request by ranisalt**
I have just stumbled across this limitation, that the walkthrough could be optionally enabled at config.lua, and done the patch :)

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -825,7 +825,7 @@ bool Player::canWalkthrough(const Creature* creature) const
 	}
 
 	const Player* player = creature->getPlayer();
-	if (!player) {
+	if (!player || g_config.getBoolean(ConfigManager::ALLOW_WALKTHROUGH)) {
 		return false;
```
Comment (gugahoa) [inline src/player.cpp]: Shouldn't be !g_config.getBoolean(ConfigManager::ALLOW_WALKTHROUGH)?

File: src/player.cpp [hunk 2]
```diff
@@ -861,7 +861,7 @@ bool Player::canWalkthroughEx(const Creature* creature) const
 	}
 
 	const Player* player = creature->getPlayer();
-	if (!player) {
+	if (!player || g_config.getBoolean(ConfigManager::ALLOW_WALKTHROUGH)) {
```
Comment (gugahoa) [inline src/player.cpp]: Same as above.

**Comment by ranisalt** [comment]
True that. Fixed.

**Comment by EvanMC** [comment]
I feel like this should wait until move events are implemented.

**Comment by marksamman** [comment]
I agree with @EvanMC.

**Comment by EPuncker** [comment]
I agree with @marksamman that agree with @EvanMC

**Comment by ranisalt** [comment]
No problem in leaving this PR open? :)

File: config.lua [hunk 3]
```diff
@@ -24,6 +24,7 @@ maxPlayers = 0
 motd = "Welcome to The Forgotten Server!"
 onePlayerOnlinePerAccount = true
 allowClones = false
+allowWalkthrough = true
```
Comment (Narzerus) [inline config.lua]: Shouldn't this be false by default? Sefaults should resemble vanilla in my opinion

Comment (ranisalt) [inline config.lua]: Nope, in Tibia players are walkthroughable by default. This is not a TFS-only feature, maybe you didn't know it has been implemented in Tibia :P

Comment (Narzerus) [inline config.lua]: WHAT? Oh hell no, sorry I got stuck in 7.4 apparently. 

Sorry man

**Pull request closed without merging**